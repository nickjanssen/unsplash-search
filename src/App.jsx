import React from "react";
import { Switch, Route, Redirect, useLocation } from "react-router-dom";
import Search from "./pages/Search";

// import unsplash from './services/unsplash';
// import { toJson } from "unsplash-js";

function App() {
  let location = useLocation();
  console.log(location.pathname);

  return (
    <Switch>
      {/* <Route path="/favorites">
      <Favorites />
    </Route> */}
      <Route path="/search">
        <Search />
      </Route>
      <Route path="/">
        <Redirect to="/search" />
      </Route>
    </Switch>
  );
}

export default App;
