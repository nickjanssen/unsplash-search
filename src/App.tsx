import React from 'react';
import './styles/main.css';

// import unsplash from './services/unsplash';
// import { toJson } from "unsplash-js";

function App() {
  React.useEffect(() => {
    const asyncFn = async () => {
      // const stream = await unsplash.search.photos("dogs", 1, 10, { orientation: "portrait" })
      // const data = await toJson(stream);
      const data = require("./exampleRes.json");

      console.log("data", data);
    };

    asyncFn();
  }, []);


  return (
    <div className="p-10 bg-red-500">
      Hello world!
    </div>
  );
}

export default App;
