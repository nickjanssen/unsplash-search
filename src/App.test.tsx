import { AppState, reducer } from "./App";

test("Can add a list", () => {
  const state: AppState = {
    lists: [],
  };

  const newState: AppState = reducer(state, {
    type: "add-list",
    list: {
      title: "Cute dogs",
      description: "Pics of dogs I adore",
      images: [],
    },
  });

  expect(newState).toMatchSnapshot();
});

test("Can add an image to a list", () => {
  const state: AppState = {
    lists: [
      {
        title: "Cute dogs",
        description: "Pics of dogs I adore",
        images: [],
      },
    ],
  };

  const newState: AppState = reducer(state, {
    type: "add-image-to-list",
    listTitle: "Cute dogs",
    image: {
      id: "U6nlG0Y5sfs",
      links: {
        download: "https://example.com/",
      },
      urls: {
        thumb:
          "https://images.unsplash.com/photo-1529472119196-cb724127a98e?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjE2MDUxMn0",
      },
      user: {
        name: "Hannah Lim",
        portfolio_url: "https://hannahlim213.github.io/portfolio-page/",
      },
    },
  });

  expect(newState).toMatchSnapshot();
});

test("Can remove an image from a list", () => {
  const state: AppState = {
    lists: [
      {
        title: "Cute dogs",
        description: "Pics of dogs I adore",
        images: [{
          id: "U6nlG0Y5sfs",
          links: {
            download: "https://example.com/",
          },
          urls: {
            thumb:
              "https://images.unsplash.com/photo-1529472119196-cb724127a98e?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjE2MDUxMn0",
          },
          user: {
            name: "Hannah Lim",
            portfolio_url: "https://hannahlim213.github.io/portfolio-page/",
          },
        }],
      },
    ],
  };

  const newState: AppState = reducer(state, {
    type: "remove-image-from-list",
    listTitle: "Cute dogs",
    imageId:  "U6nlG0Y5sfs"
  });

  expect(newState).toMatchSnapshot();
});
