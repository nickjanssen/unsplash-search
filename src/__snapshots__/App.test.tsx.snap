// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can add a list 1`] = `
Object {
  "lists": Array [
    Object {
      "description": "Pics of dogs I adore",
      "images": Array [],
      "title": "Cute dogs",
    },
  ],
}
`;

exports[`Can add an image to a list 1`] = `
Object {
  "lists": Array [
    Object {
      "description": "Pics of dogs I adore",
      "images": Array [
        Object {
          "id": "U6nlG0Y5sfs",
          "links": Object {
            "download": "https://example.com/",
          },
          "urls": Object {
            "thumb": "https://images.unsplash.com/photo-1529472119196-cb724127a98e?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=200&fit=max&ixid=eyJhcHBfaWQiOjE2MDUxMn0",
          },
          "user": Object {
            "name": "Hannah Lim",
            "portfolio_url": "https://hannahlim213.github.io/portfolio-page/",
          },
        },
      ],
      "title": "Cute dogs",
    },
  ],
}
`;

exports[`Can remove an image from a list 1`] = `
Object {
  "lists": Array [
    Object {
      "description": "Pics of dogs I adore",
      "images": Array [],
      "title": "Cute dogs",
    },
  ],
}
`;
